ui/app/search/search-ctrl.js

```diff
@@ -26,30 +26,22 @@
         var oldParams = mlSearch.getParams();
         var newParams = mlSearch.getCurrentParams();
 
-        // restore params, if we have cached results, and no params provided
-        if (model.search.results && _.isEqual( {}, newParams )) {
+        // capture URL params in mlSearch and ctrl model
+        mlSearch.fromParams().then(function() {
 
-          $location.search( oldParams );
```

Comment:
I think that was the intention, to trigger locationChangeSuccess. That has an extra check to only do an actual search if url changed. You think that check was flawed?